src/sentry/workflow_engine/endpoints/organization_detector_workflow_details.py

```diff
@@ -0,0 +1,95 @@
+from drf_spectacular.utils import extend_schema
+from rest_framework import status
+from rest_framework.exceptions import PermissionDenied
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry import audit_log
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases import OrganizationEndpoint
+from sentry.api.bases.organization import OrganizationDetectorPermission
+from sentry.api.exceptions import ResourceDoesNotExist
+from sentry.api.serializers import serialize
+from sentry.apidocs.constants import (
+    RESPONSE_BAD_REQUEST,
+    RESPONSE_FORBIDDEN,
+    RESPONSE_NOT_FOUND,
+    RESPONSE_UNAUTHORIZED,
+)
+from sentry.apidocs.parameters import DetectorWorkflowParams, GlobalParams
+from sentry.deletions.models.scheduleddeletion import RegionScheduledDeletion
+from sentry.models.organization import Organization
+from sentry.utils.audit import create_audit_entry
+from sentry.workflow_engine.endpoints.serializers import DetectorWorkflowSerializer
+from sentry.workflow_engine.endpoints.validators.detector_workflow import can_edit_detector
+from sentry.workflow_engine.models.detector_workflow import DetectorWorkflow
+
+
+@region_silo_endpoint
+class OrganizationDetectorWorkflowDetailsEndpoint(OrganizationEndpoint):
+    def convert_args(self, request: Request, detector_workflow_id, *args, **kwargs):
+        args, kwargs = super().convert_args(request, *args, **kwargs)
+        try:
+            kwargs["detector_workflow"] = DetectorWorkflow.objects.get(
+                workflow__organization=kwargs["organization"], id=detector_workflow_id
+            )
+        except DetectorWorkflow.DoesNotExist:
+            raise ResourceDoesNotExist
+
+        return args, kwargs
+
+    publish_status = {
+        "GET": ApiPublishStatus.EXPERIMENTAL,
+        "DELETE": ApiPublishStatus.EXPERIMENTAL,
+    }
+    owner = ApiOwner.ISSUES
+    permission_classes = (OrganizationDetectorPermission,)
+
+    @extend_schema(
+        operation_id="Fetch a DetectorWorkflow",
+        parameters=[
+            GlobalParams.ORG_ID_OR_SLUG,
+            DetectorWorkflowParams.DETECTOR_WORKFLOW_ID,
+        ],
+        responses={
+            201: DetectorWorkflowSerializer,
+            400: RESPONSE_BAD_REQUEST,
+            401: RESPONSE_UNAUTHORIZED,
+            403: RESPONSE_FORBIDDEN,
+            404: RESPONSE_NOT_FOUND,
+        },
+    )
+    def get(
+        self, request: Request, organization: Organization, detector_workflow: DetectorWorkflow
+    ):
+        """
+        Returns a DetectorWorkflow
+        """
+        serialized_detector_workflow = serialize(
+            detector_workflow,
+            request.user,
+            DetectorWorkflowSerializer(),
+        )
+        return Response(serialized_detector_workflow)
```

Comment:
just curious -- where in the UI are we planning to get a single detector workflow? (is there a separate list view for this?)

Reply:
This isn't being used anywhere in the UI now, it just felt weird only having the DELETE 😅